From 6a54e9b7f1b898b0c51b8c871e4df39cc42344eb Mon Sep 17 00:00:00 2001
From: Alexander Martinz <amartinz@shiftphones.com>
Date: Thu, 12 Apr 2018 17:25:06 +0200
Subject: [PATCH 2/5] Icons: wrap all legacy icons to adaptive icons

Previously only apps targetting API 26 (O) would get wrapped as
adaptive icon, resulting in legacy icons still looking out of place.

Wrap all icons, independent of target API.

Change-Id: I3647d95c7f4f5c0515a14cbf4eebd5afcdfb38ae
Signed-off-by: Alexander Martinz <amartinz@shiftphones.com>
---
 src/com/android/launcher3/IconCache.java | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/src/com/android/launcher3/IconCache.java b/src/com/android/launcher3/IconCache.java
index f1f495047..3230e70d3 100644
--- a/src/com/android/launcher3/IconCache.java
+++ b/src/com/android/launcher3/IconCache.java
@@ -370,7 +370,7 @@ public class IconCache {
         if (entry == null) {
             entry = new CacheEntry();
             entry.icon = LauncherIcons.createBadgedIconBitmap(getFullResIcon(app), app.getUser(),
-                    mContext,  app.getApplicationInfo().targetSdkVersion);
+                    mContext, Build.VERSION_CODES.O);
         }
         entry.title = app.getLabel();
         entry.contentDescription = mUserManager.getBadgedLabelForUser(entry.title, app.getUser());
@@ -571,7 +571,7 @@ public class IconCache {
                 if (info != null) {
                     entry.icon = LauncherIcons.createBadgedIconBitmap(
                             getFullResIcon(info), info.getUser(), mContext,
-                            infoProvider.get().getApplicationInfo().targetSdkVersion);
+                            Build.VERSION_CODES.O);
                 } else {
                     if (usePackageIcon) {
                         CacheEntry packageEntry = getEntryForPackageLocked(
@@ -677,7 +677,8 @@ public class IconCache {
                     // Load the full res icon for the application, but if useLowResIcon is set, then
                     // only keep the low resolution icon instead of the larger full-sized icon
                     Bitmap icon = LauncherIcons.createBadgedIconBitmap(
-                            appInfo.loadIcon(mPackageManager), user, mContext, appInfo.targetSdkVersion);
+                            appInfo.loadIcon(mPackageManager), user, mContext,
+                            Build.VERSION_CODES.O);
                     if (mInstantAppResolver.isInstantApp(appInfo)) {
                         icon = LauncherIcons.badgeWithDrawable(icon,
                                 mContext.getDrawable(R.drawable.ic_instant_app_badge), mContext);
@@ -814,7 +815,7 @@ public class IconCache {
     }
 
     private static final class IconDB extends SQLiteCacheHelper {
-        private final static int DB_VERSION = 18;
+        private final static int DB_VERSION = 19;
 
         private final static int RELEASE_VERSION = DB_VERSION +
                 (FeatureFlags.LAUNCHER3_DISABLE_ICON_NORMALIZATION ? 0 : 1);
-- 
2.17.0

